orttraining/orttraining/python/training/ortmodule/__init__.py

```diff
@@ -74,6 +76,17 @@ def override_torch_cuda_manual_seed(seed):
 torch_cuda_manual_seed = torch.cuda.manual_seed
 torch.cuda.manual_seed = override_torch_cuda_manual_seed
 
+
+def use_deterministic_algorithms(enabled):
```

Comment:
Shall we only keep this internal for now and not allow users to enable/disable this for the time being (until we feel the need to explicitly allow users to change this).

Reply:
onnxruntime repo is internal :) only torch_ort is public

Reply:
My concern (for the time being) is not with the API but with how we are currently using it. We create a new execution agent whenever `torch.are_deterministic_algorithms_enabled()` is different from `ortmodule.are_deterministic_algorithms_enabled()`. With this being exposed, users may start using this API and unknowingly (by not setting the same on torch) trigger re-creation of the execution agent (since our logic is checking this value againts torch's).

Reply:
Shall we instead just simply use `torch.are_deterministic_algorithms_enabled()` all the time instead of defining our own `are_deterministic_algorithms_enabled()`? What advantage does this bring us at this time to expose this if we anyway only reading the torch's implementation to set our determinism.

Reply:
We keep our own to detect changes to determinism state. From one forward call to the next, if we only use pytorch's, how can we know pytorch change from true to false and otherwise? The idea is that our flag is set during initialization and then always compared to pytorch's for equality. Any mismatch leads to recreation of execution session. Makes sense?

Reply:
For your previous concern, I think you are right. If we expose it to the public `torch_ort` API, then the user could accidentally cause the execution session creation. I will append `_` to them until we decide about its utility as standalone APIs

Reply:
Yeah, we need a way to determine change in pytorch determinism. And my thought was to persist the setting that we used to save the `SessionOptions` determinism and then compare against that so we don't need to keep a gloabl variable which could go out of sync (particularly if we always want to follow pytorch's determinism).
Let's go ahead and merge this now. And we can revisit this later if we feel that this causes confusion (or consistency issues).